sign.go

```diff
@@ -100,6 +100,20 @@ func (sk *SecretKey) Sign(t *merlin.Transcript) (*Signature, error) {
 	}, nil
 }
 
+// Sign uses the schnorr signature algorithm to sign a message
+// See the following for the transcript message
+// https://github.com/w3f/schnorrkel/blob/db61369a6e77f8074eb3247f9040ccde55697f20/src/sign.rs#L158
+// Schnorr w/ transcript, secret key x:
+// 1. choose random r from group
+// 2. R = gr
+// 3. k = scalar(transcript.extract_bytes())
+// 4. s = kx + r
+// signature: (R, s)
+// public key used for verification: y = g^x
+func (kp *Keypair) Sign(t *merlin.Transcript) (*Signature, error) {
+	return kp.secretKey.Sign(t)
```

Comment:
should you check if `secretKey` is not `nil` before calling the `Sign`? I'm asking because `Keypair` is exported and can be created using `&Keypair{}`

Reply:
yup good call!